### PR TITLE
add dns keys datasource

### PR DESCRIPTION
**Pull request by megan07**
If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/third_party/terraform/utils/framework_provider.go.erb [hunk 1]
```diff
@@ -114,16 +114,84 @@ func (p *frameworkProvider) Schema(_ context.Context, _ provider.SchemaRequest,
...
+            "iam_credentials_custom_endpoint": &schema.StringAttribute{
+                Optional:     true,
+                // ValidateFunc: validateCustomEndpoint,
+            },
+            "resource_manager_v3_custom_endpoint": &schema.StringAttribute{
+                Optional:     true,
+                // ValidateFunc: validateCustomEndpoint,
+            },
+            <% unless version == "ga" -%>
+            "runtimecconfig_custom_endpoint": &schema.StringAttribute{
```
[the remark this answers is not in the document]
Comment (SarahFrench) [reply mmv1/third_party/terraform/utils/framework_provider.go.erb:182]: I'm reviewing this PR but this change is relevant to https://github.com/GoogleCloudPlatform/magic-modules/pull/7182 and https://github.com/GoogleCloudPlatform/magic-modules/pull/7169

File: mmv1/third_party/terraform/data_sources/data_source_dns_record_set.go [hunk 2]
```diff
@@ -1,86 +1,145 @@
...
+	Project     types.String `tfsdk:"project"`
+}
 
-			"ttl": {
-				Type:     schema.TypeInt,
-				Computed: true,
-			},
+func (d *GoogleDnsRecordSetDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_dns_record_set"
+}
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/data_sources/data_source_dns_record_set.go:40]: I think this is a source of an issue.

TypeName needs to be "the full data source type, including the provider type prefix and an underscore. For example, examplecloud_thing." ([from docs](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework@v1.1.1/datasource#MetadataResponse)).

Even when using the google-beta provider the types still only start with `google_` not `google-beta_`

However, the value of `req.ProviderTypeName` is "google-beta" so the provider is sending a `datasource.MetadataResponse` with `google-beta_dns_record_set` as the type name

Bet that's causing an issue somewhere, but I'm not sure where!

Comment (SarahFrench) [reply mmv1/third_party/terraform/data_sources/data_source_dns_record_set.go:40]: I guess in the `DataSources` function that lists out the data sources implemented by the provider is using that name, hence this error when running acceptance tests:

> | Error: Invalid data source
  | 
  |   on terraform_plugin_test.tf line 17, in data "google_dns_record_set" "rs":
  |   17: data "google_dns_record_set" "rs" {
  | 
  | The provider hashicorp/google does not support data source
  | "google_dns_record_set".

File: mmv1/third_party/terraform/data_sources/data_source_dns_keys.go [hunk 3]
```diff
@@ -1,229 +1,436 @@
...
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Get the DNSKEY and DS records of DNSSEC-signed managed zones",
+
+		Attributes: map[string]schema.Attribute{
+			"managed_zone": schema.StringAttribute{
+				Description:         "The Name of the zone.",
+				MarkdownDescription: "The Name of the zone.",
+				Required:            true,
+				// 				DiffSuppressFunc: compareSelfLinkOrResourceName,
```
[the remark this answers is not in the document]
Comment (megan07) [reply mmv1/third_party/terraform/data_sources/data_source_dns_keys.go]: as discussed on our call, it seems we have parity. it looks like the data source was never able to be "created" with the short managed zone name, however, the diff would be suppressed between long name and short name, and that stands true with this as well.

**Review by SarahFrench — APPROVED** [review]
LGTM :shipit:

**Pull request merged**